tools/ci/production-upgrade

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+# This test is for upgrading the production server
+set -e
+set -x
+
+# Uninstall rabbitmq-server and supervisor as this has configurations generated for previous containers
+# We install it again from scratch.
+sudo apt-get remove rabbitmq-server supervisor && sudo apt-get purge rabbitmq-server supervisor
+sudo apt-get install rabbitmq-server supervisor
+
+# Start the postgresql service.
+sudo service postgresql start
+
+# Starting the rabbitmq-server
+if ! sudo service rabbitmq-server start; then
+    echo
+    echo
+    echo "Starting rabbitmq-server failed. Trying again:"
+    sudo service rabbitmq-server start
+fi
+
+# Applying puppet changes. This also regenerates the supervisor configurations.
+sudo /home/zulip/deployments/current/scripts/zulip-puppet-apply -f
+
+# Stopping nginx service started by above command.
+# This causes unexpected `Unable to stop Service[nginx]` error in the puppet apply step of upgrade otherwise.
+if ! sudo service nginx stop; then
+    echo
+    echo
+    echo "Stoping nginx failed. Trying again:"
+    sudo service nginx stop
+fi
+
+# Perform the upgrade.
+if sudo /home/zulip/deployments/current/scripts/upgrade-zulip /tmp/zulip-server-test.tar.gz; then
+    echo
+    echo
+    echo "Upgrade of production server complete!"
+    echo "Starting the verification step soon."
+    # Required to let zulip workers start properly.
+    supervisorctl status
+    sleep 120
+    supervisorctl status
+    exit 0
+fi
```

Comment:
Is this 120s sleep required, or just a relic from testing of previous versions?  It seems like a really long time, and in any case we probably want to instead do some sort of "wait until `curl {zulip_server_url}` gets a 200 or 500, up to 120s. (or something) so that we don't waste a ton of time.